### PR TITLE
Nerfs harm/disarm steppies

### PR DESCRIPTION
**Pull request by Novacat**
- Makes bad steppies have a chance to miss on mobs not on help intent
- Removes halloss and replaces with weaken
- Steppies now only forcerests people on help intent

### CODE REVIEW

File: code/modules/vore/resizing/resize_vr.dm [hunk 1]
```diff
@@ -246,7 +246,7 @@ var/const/RESIZE_A_SMALLTINY = (RESIZE_SMALL + RESIZE_TINY) / 2
 	now_pushing = 0
 	forceMove(tmob.loc)
 	if(a_intent == I_GRAB || a_intent == I_DISARM)
-		tmob.resting = 1
+		tmob.Weaken(1)
```
Comment (Rykka-Stormheart) [inline code/modules/vore/resizing/resize_vr.dm:249]: Weaken(2) or Weaken(3) would be better, imo

**Comment by Screemonster** [comment]
Honestly? Maybe shouldn't even be an auto-weaken if the victim also has a hostile (disarm, harm, grab) intent. Disarm-intent clicks don't automatically succeed either.

**Comment by Rykka-Stormheart** [comment]
> something to the effect of the following added to the block in lines 236-245 should do the trick:
> 
> if(tmob.a_intent != I_HELP && prob(50))
> to_chat(pred, "[prey] dodges out from under your foot!")
> to_chat(prey, "You narrowly avoid [pred]'s foot!")
> return FALSE

Agreed. Macro stomps shouldn't be an instant win, though imo I'd reduce it from prob(50) to something like prob(35/45). Still not an instant win, but not a hard 50/50.

**Comment by Screemonster** [comment]
> Agreed. Macro stomps shouldn't be an instant win, though imo I'd reduce it from prob(50) to something like prob(35/45). Still not an instant win, but not a hard 50/50.

honestly I just pulled "50" out of my ass because I couldn't be bothered looking up what the actual odds of a successful disarm are

**Comment by Rykka-Stormheart** [comment]
> > Agreed. Macro stomps shouldn't be an instant win, though imo I'd reduce it from prob(50) to something like prob(35/45). Still not an instant win, but not a hard 50/50.
> 
> honestly I just pulled "50" out of my ass because I couldn't be bothered looking up what the actual odds of a successful disarm are. Note that in that code, the prob would be the odds of the micro _dodging_, not the macro _hitting_ so pick the number accordingly.

Ye, I still think it should be prob(35) or prob(45). Macros are large and getting into close combat can often be a death sentence because of how hard it is to click shit near you.

**Comment by MrSign** [comment]
Yeah, if we could implement a 'dodge' mechanic or something similar to disarm where it has a chance of failing, that'd be great.

**Comment by lbnesquik** [comment]
Muh coils stuff

**Comment by Rykka-Stormheart** [comment]
> Yeah, if we could implement a 'dodge' mechanic or something similar to disarm where it has a chance of failing, that'd be great.

That's been implemented!

**Comment by KasparoVy** [comment]
Man I must be unrobust. Not sure of how to dodge aside from moving my dude, flipping like a gymnast or using jump boots to dive into a wall

File: code/modules/vore/resizing/resize_vr.dm [hunk 2]
```diff
@@ -243,10 +243,18 @@ var/const/RESIZE_A_SMALLTINY = (RESIZE_SMALL + RESIZE_TINY) / 2
...
+	if(tmob.a_intent != I_HELP && prob(35))
+		to_chat(pred, "<span class='danger'>[prey] dodges out from under your foot!</span>")
+		to_chat(prey, "<span class='danger'>You narrowly avoid [pred]'s foot!</span>")
+		return FALSE
+
 	now_pushing = 0
 	forceMove(tmob.loc)
 	if(a_intent == I_GRAB || a_intent == I_DISARM)
-		tmob.resting = 1
+		if(tmob.a_intent = I_HELP)
```
Comment (Meghan-Rossi) [inline code/modules/vore/resizing/resize_vr.dm:254]: This is giving an "Assignment in condition" warning, and I think == would make more sense here.

**Comment by lbnesquik** [comment]
REEEE

**Pull request merged**